csvcubed/csvcubed/cli/inspect/inspectdatasetmanager.py

```diff
@@ -0,0 +1,299 @@
+    ComponentPropertyAttributeURI,
+    ComponentPropertyType,
+)
+from csvcubed.models.inspectdataframeresults import (
+    DatasetObservationsByMeasureUnitInfoResult,
+    DatasetObservationsInfoResult,
+)
+from csvcubed.utils.csvdataset import CanonicalShapeRequiredCols
+
+_logger = logging.getLogger(__name__)
+
+
+class DatasetMeasureType(Enum):
+    """
+    The type of dataset measure.
+    """
+
+    SINGLE_MEASURE = auto()
+
+    MULTI_MEASURE = auto()
+
+
+class DatasetUnitType(Enum):
+    """
+    The type of dataset unit.
+    """
+
+    SINGLE_UNIT = auto()
+
+    MULTI_UNIT = auto()
+
+
+def _filter_components_from_dsd(
+    graph: Graph, dsd_uri: URIRef, json_path: Path, field: ComponentField, filter: str
+) -> List[QubeComponentResult]:
+    """
+    Filters the components for the given filter.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `List[QubeComponentResult]` - filtered results.
+    """
+    result_qube_components: QubeComponentsResult = select_csvw_dsd_qube_components(
+        graph, dsd_uri, json_path
+    )
+
+    return [
+        component
+        for component in result_qube_components.qube_components
+        if component.as_dict()[field.value] == filter
+    ]
+
+
+def load_csv_to_dataframe(json_path: Path, csv_path: Path) -> DataFrame:
+    """
+    Loads the csv in given path to a Panda Dataframe.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DataFrame` - Dataframe of the csv.
+    """
+
+    try:
+        dataset_path = json_path.parent / csv_path
+        _logger.debug(f"Dataset path: {dataset_path.absolute()}")
+
+        dataset = DataFrame(pd.read_csv(dataset_path))
+        _logger.info("Successfully loaded csv into dataframe.")
+        return dataset
+    except Exception as ex:
+        raise Exception("An error occured while loading csv into dataframe.") from ex
+
+
+def get_dataset_measure_type(
+    graph: Graph, dsd_uri: URIRef, json_path: Path
+) -> DatasetMeasureType:
+    """
+    Detects whether the dataset is single-measure or multi-measure.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetMeasureType`
+    """
+    filtered_components = _filter_components_from_dsd(
+        graph,
+        dsd_uri,
+        json_path,
+        ComponentField.PropertyType,
+        ComponentPropertyType.Measure.value,
+    )
+    measure_type = (
+        DatasetMeasureType.MULTI_MEASURE
+        if len(filtered_components) > 1
+        else DatasetMeasureType.SINGLE_MEASURE
+    )
+    _logger.debug(f"Dataset measure type: {measure_type}")
+
+    return measure_type
+
+
+def get_dataset_unit_type(dataset: DataFrame) -> DatasetUnitType:
+    """
+    Detects whether the dataset is single-unit or multi-unit.
+
+    Member of :file:`./inspectdatasetmanager.py`
+
+    :return: `DatasetUnitType`
+    """
+    unit_type = (
+        DatasetUnitType.MULTI_UNIT
+        if len(dataset[CanonicalShapeRequiredCols.Unit.value].unique()) > 1
```

Comment:
The units column doesn't necessarily have this name. I'd say the safest check to make here is to look at the data structure components returned from your SPARQL query and see whether one of them is a units column. If it is, then it's shaped in a multi-unit fashion.